src/App.tsx

```diff
@@ -1,35 +1,181 @@
+  );
+  const [secondTeamError, setSecondTeamError] = useState<string>(
+    "Input can not be empty!"
+  );
+  const [tickets, setTickets] = useState<string>("");
+  const [ticketsDirty, setTicketsDirty] = useState<boolean>(false);
+  const [ticketsError, setTicketsError] = useState<string>(
+    "Input can not be empty!"
+  );
+  const [stadium, setStadium] = useState<string>("");
+  const [stadiumDirty, setStadiumDirty] = useState<boolean>(false);
+  const [stadiumError, setStadiumError] = useState<string>(
+    "Input can not be empty!"
+  );
+  const [formValid, setFormValid] = useState<boolean>(false);
+  const [isVisible, setIsVisible] = useState<boolean>(false);
+  const [blocks, setBlocks] = useState<Block[]>([]);
+
+  useEffect(() => {
+    if (firstTeamError || secondTeamError || ticketsError || stadiumError) {
+      setFormValid(false);
+    } else {
+      setFormValid(true);
+    }
+  }, [firstTeamError, secondTeamError, ticketsError, stadiumError]);
+
+  const teamHandler = (
+    e: React.ChangeEvent<HTMLInputElement>,
+    setTeam: React.Dispatch<React.SetStateAction<string>>,
+    setDirty: React.Dispatch<React.SetStateAction<boolean>>,
+    setError: React.Dispatch<React.SetStateAction<string>>
+  ) => {
+    const value = e.target.value;
+    setTeam(value);
+    const invalidChars = /[!@#$%^&*(),.?":{}|<>]/;
+    setDirty(true);
+    if (invalidChars.test(value)) {
+      setError("The name of the team must consist only of letters or numbers!");
+    } else {
+      setError("");
+    }
+  };
+
+  const ticketsHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+    setTickets(value);
+    setTicketsDirty(true);
+    if (parseInt(value) < 0) {
+      setTicketsError("Quantity of tickets can not be negative!");
+    } else {
+      setTicketsError("");
+    }
+  };
+
+  const stadiumHandler = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const value = e.target.value;
+    setStadium(value);
+    setStadiumDirty(true);
+    if (!value) {
+      setStadiumError("Please, choose the stadium!");
+    } else {
+      setStadiumError("");
+    }
+  };
+
+  const blurHandler = (
+    e: React.FocusEvent<HTMLInputElement | HTMLSelectElement>
+  ) => {
+    switch (e.target.name) {
+      case "firstTeam":
+        setFirstTeamDirty(true);
+        if (!firstTeam) {
+          setFirstTeamError("Input can not be empty!");
+        }
+        break;
+      case "secondTeam":
+        setSecondTeamDirty(true);
+        if (!secondTeam) {
+          setSecondTeamError("Input can not be empty!");
+        }
+        break;
+      case "tickets":
+        setTicketsDirty(true);
+        if (!tickets) {
+          setTicketsError("Input can not be empty!");
+        }
+        break;
+      case "stadium":
+        setStadiumDirty(true);
+        if (!stadium) {
+          setStadiumError("Please, choose the stadium!");
+        }
+        break;
+    }
+  };
+
+  const addButtonHandler = () => {
```

Comment:
Also, it would be better to encapsulate all form states in Form component. You can move all handlers as well. Then you will pass addButtonHandler to Form but this method will take params firstteam, secontteam, tickets.... and you can call it from Form and pass all needed information